src/components/Index.tsx

```diff
@@ -78,6 +80,52 @@ const Index: React.FC = () => {
     return data;
   };
 
+  const fetchBirdData: FetchBirdData = async (birdName) => {
+    try {
+      // First try with bird name
+      const response = await fetch(
+        `https://en.wikipedia.org/api/rest_v1/page/summary/${birdName}`
+      );
+      if (response.ok) {
+        const data = await response.json();
+        if (data.type !== "disambiguation") {
+          return data;
+        }
```

Comment:
this function is overly complex as it tries to handle multiple responsibilities. It should be simplified to focus solely on fetching data and returning it, without making multiple calls to different endpoints or performing unrelated logic.